src/server/kernel/wxdgsurface.cpp

```diff
@@ -287,6 +295,32 @@ WXdgSurface *WXdgSurface::parentXdgSurface() const
         if (!parent)
             return nullptr;
         return fromHandle(QWXdgToplevel::from(parent));
+    } else if (auto popup = d->handle->toPopup()) {
```

Comment:
Why

Reply:
`parentXdgSurface` of Popup is still useful in  https://github.com/vioken/waylib/blob/master/examples/tinywl/StackWorkspace.qml#L106-L117, should not remove